themes/default/content/blog/launch-redshift-into-aws-vpc-with-pulumi/index.md

```diff
@@ -0,0 +1,301 @@
+                Resource: "*",
+            }],
+        }),
+    }]
+});
+```
+
+Place those declarations after the statements related to the AWS S3 Bucket but before the statements declaring the Amazon Redshift cluster.
+
+We declare the AWS IAM role similar to the AWS console. We use a JSON-formatted object containing version and statement attributes. As with all other AWS services, the version is the publication date. The statement contains the action, effect, and principal. Here, we set those to `sts:AssumeRole`, `Allow`, and `redshift.amazonaws.com`, respectively. Then, we define the inline policy, which enables the principal to have read-only access to Amazon S3.
+
+Deploy those resources by typing:
+
+```bash
+$ pulumi up
+```
+
+We can then see the new role under the IAM console:
+
+![Screenshot of the AWS console showing the newly provisioned IAM role](./aws-console-6-iam-role.png)
+
+## Updating the Amazon Redshift Cluster with the AWS IAM Role
+
+In the last step, we assign the AWS IAM role to the Amazon Redshift Cluster. We add the `iamRoles` attribute to the Amazon Redshift cluster declaration:
+
+```typescript
+// Redshift cluster
+const redshiftCluster = new aws.redshift.Cluster("pulumi-redshift-cluster", {
+    clusterIdentifier: "pulumi-redshift-cluster",
+    clusterType: "single-node",
+    databaseName: "tempdb",
+    masterPassword: "Passw0rD",
+    masterUsername: "admin",
+    nodeType: "dc2.large",
+    clusterSubnetGroupName: subnetGroup.name,
+    skipFinalSnapshot: true,
+    iamRoles: [iamRole.arn] // <-- Add this property.
+});
+```
+
+Specify the roles using their Amazon Resource Names (ARNs). Here, we use just one role and obtain its ARN by reading the arn property of the object created previously.
+
+We can now update the cluster to confirm that we assigned the AWS IAM role to the cluster:
+
+```bash
+$ pulumi up
+```
+
+To see the IAM role, go to the Properties tab of the cluster and scroll down to Associated IAM roles:
+
+![Screenshot of the AWS console showing the IAM role associated with the Redshift cluster](./aws-console-7-cluster-iam-roles.png)
+
+After preparing the infrastructure, we can now use it to process data. The first step is to load data from the S3 source, [as demonstrated in the AWS documentation](https://docs.aws.amazon.com/redshift/latest/dg/tutorial-loading-data.html).
+
+Now remove all deployed cloud resources using the single command:
+
+```bash
+pulumi destroy
+```
+
+![CLI output of the update showing the cluster and all resources now removded](./destroy.png)
+
+## Summary
+
+This article demonstrated how to use Pulumi to declare and provision several AWS resources using IaC. It started with the Pulumi AWS project template. It then updated it to provision a virtual private cloud, Internet gateway, subnet, subnet group, S3 bucket, IAM role, and the Amazon Redshift cluster. Doing so achieved a declaration of cloud resources, which we can keep in the git repository and track all changes.
```

Comment:
i would probably talk about the next steps which is here is the URL string to access the cluster. You can hook up your app to dump data into it. You can also hook up a BI tool to query the data, etc.

Reply:
Yep, :100: agree -- [this was in my original feedback](https://docs.google.com/document/d/1qllVyNGk0sUg3EUxf-gaggdPBWEPMTzgCWUq4VQGe_E/edit?disco=AAAAfXc9_T0) as well.